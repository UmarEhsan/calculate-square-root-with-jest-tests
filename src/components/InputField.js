import React, {useState} from 'react';
const InputField = ({ handleChange }) => {
    const [squareRoot, setSquareRoot] = useState(0); 
    const handleInputChange = (event) => {
        event.preventDefault();
        const value = parseInt(event.target.value);
        setSquareRoot(value);
        handleChange(value);
    };


    return (
        <div style={{ display: "inline" }}>
          
          <input
            className="form-control"
            type="number"
            onChange={handleInputChange}
            name='square-root'
            value={squareRoot.toString()}
          />
         </div>
      );
};

export default InputField;