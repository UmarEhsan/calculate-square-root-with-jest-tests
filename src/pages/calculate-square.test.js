import React from 'react';
import { render } from '@testing-library/react';
import CalculateSquareRoot from '../pages/calculate-square';
import InputField from '../components/InputField';
import TestRenderer  from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from 'enzyme';
import calulateSquare from '../constants/calculateSquareApi';
import axios from 'axios';

configure({adapter: new Adapter()});

it('render correctly text component', () => {  
    const TextInputComponent = TestRenderer.create(<InputField />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});


it('check the input value to be 0 at the start', () => {
    const wrapper = mount(<CalculateSquareRoot />)
    expect(wrapper.find('input[type="number"]').prop('value')).toEqual(String(0));
});

it(`sould test the api call`, async () => {
        jest.setTimeout(10000);
        const result = await calulateSquare(16);
        expect(result.title).toBe(4);
    });
