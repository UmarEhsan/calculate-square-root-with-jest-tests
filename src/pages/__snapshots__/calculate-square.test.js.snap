// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
<div
  style={
    Object {
      "display": "inline",
    }
  }
>
  <input
    className="form-control"
    name="square-root"
    onChange={[Function]}
    type="number"
    value="0"
  />
</div>
`;
